src/contributing/upstreaming.md

```diff
@@ -0,0 +1,77 @@
+# Code upstreaming guide
+
+This page presents the details around upstreaming code in Parsec-related repositories. Currently our
+repos are mainly Rust crates, so this guide is accordingly geared towards them - when other
+upstreaming guidelines become relevant, e.g. specific to Go libraries, new sections should be added
+below.
+
+The following points are relevant to all software we release or upstream:
+
+- For upstreamed versions of a project a tag carrying the version number should be upstreamed to
+   GitHub, e.g. if version 2.3.1 was upstreamed, a tag named "2.3.1" must be added to the commit in
+   GitHub.
+- If multiple upstreamable projects reside in the same repo, the tags should reflect both the
+   project and the version number, e.g. if version 2.3.1 of project X was upstreamed, a tag named
+   "X-2.3.1" or similar must be used.
+- Projects should follow [Semantic Versioning](https://semver.org/) rules.
+- Following the release of a new version of publicly-facing projects, update the `CHANGELOG.md`
+   file. For some projects it's also advised to create a GitHub Release from the tag mentioned
+   above.
+
+## Rust crates
+
+The process around publishing Rust crates is tightly linked with crates.io. In order to publish or
+have control over crates, you must own an account on the website.
+
+- From the main page of crates.io you can log in using your GitHub account.
+- Go to your account settings page and generate a new API token - this will be used on the command
+   line for publishing the crates.
+- Using the token generated above, log in on the command line: `cargo login <token>`.
```

Comment:
Maybe these steps can be replaced linking the instructions from [this page](https://doc.rust-lang.org/cargo/reference/publishing.html) (section Before you first publish)?

Reply:
Dun? I didn't link that section, I think reading the whole page is a good idea anyway